pyspinw/structures.py

```diff
@@ -0,0 +1,183 @@
+""" Magnetic structures """
+
+
+import numpy as np
+
+from pyspinw.site import LatticeSite
+from pyspinw.symmetry.group import SpaceGroup, MagneticSpaceGroup, SymmetryGroup
+from pyspinw.symmetry.supercell import Supercell
+from pyspinw.symmetry.unitcell import UnitCell
+from pyspinw.tolerances import tolerances
+from pyspinw.util import connected_components, arraylike_equality
+
+
+class Structure:
+    """ Representation of the magnetic structure """
+
+    def __init__(self,
+                 sites: list[LatticeSite],
+                 spacegroup: SymmetryGroup,
+                 unit_cell: UnitCell,
+                 supercell: Supercell):

```

Comment:
Could we make this type `supercell: Supercell | None = None` where if it is not given then line 26 constructs a `TrivialSupercell()` object with no scaling? (This would be like the Matlab SpinW default of `nExt=[1 1 1]` when [constructing an empty spinw object](https://github.com/SpinW/spinw/blob/dc06a4bec2c44bcfde6baa630ecb6329fb4b68ba/swfiles/%40spinw/private/datastruct.m#L46).)